### PR TITLE
Extract pixel minimap into a separate class, clean the code and improve its performance

### PR DESCRIPTION
**Pull request by codemime**
#### Summary

```SUMMARY: Performance "Extract pixel minimap into a separate class, clean the code and improve its performance"```

#### Purpose of change

It's a preparation for #21951 and also partial debloating `cata_tiles` . I split the PR on two because it was too large as a single piece. This is the first part.

#### Describe the solution

- Routine extraction of the new class (see bd9549d and a63f883);
- Decoupling it from the `options` singleton (see 552bcd3);
- More efficient cache population. There were 132 x 132 = 17424 `std::map` lookups (the nested loops over each tile of the `map`) during the cache population phase. Now there's only 11 x 11 = 121 due to the rearrangements made there (see efc84ee);
- A bit more efficient drawing. There were 121 map lookups during the drawing phase, now there's none (see 4ef9074);
- Various other code improvements in the rest of the commits (see their descriptions for details).

#### Describe alternatives you've considered

Getting rid of the `std::map` there entirely. This is doable, but I decided to leave it for now: 121 lookups per drawing cycle are tolerable.

#### Additional context

The performance gain was measured by means of the embedded draw benchmark.

Before:
before

After:
after

The second part containing the actual feature will follow.

### CODE REVIEW

**Comment by ifreund** [comment]
Looks like the curses build failed on travis because of missing `SDL.h`.  Guess you need an `#ifdef TILES` somewhere in `pixel_minimap.h`?

**Comment by codemime** [comment]
You were right, thanks. I forgot about the preprocessor conditionals.

File: src/sdl_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    if( alt_rect_tex_enabled ) {
+        SetTextureColorMod( alt_rect_tex, r, g, b );
+        RenderCopy( renderer, alt_rect_tex, NULL, &rect );
+    } else {
+        SetRenderDrawColor( renderer, r, g, b, 255 );
+        RenderFillRect( renderer, &rect );
+    }
+}
+
+#endif // SDL_TILES
```
Comment (ZhilkinSerg) [inline src/sdl_utils.cpp]:
```suggestion
#endif // SDL_TILES

```

File: src/pixel_minimap.cpp [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+        for( int y = -rect.h + std::abs( x ), y_max = rect.h - std::abs( x ); y <= y_max; ++y ) {
+            const int divisor = 2 * ( ( std::abs( y ) == rect.h - std::abs( x ) ) ? 1 : 0 ) + 1;
+
+            SetRenderDrawColor( renderer, color.r / divisor, color.g / divisor, color.b / divisor, 0xFF );
+            RenderDrawPoint( renderer, rect.x + x, rect.y + y );
+        }
+    }
+}
+
+#endif // SDL_TILES
```
Comment (ZhilkinSerg) [inline src/pixel_minimap.cpp]:
```suggestion
#endif // SDL_TILES

```

**Pull request merged**